### PR TITLE
subtype: fast path for Type == TypeVar

### LINKED ISSUE #56606: Extremely slow subtype query causes unreasonable compilation times
```ruby
struct Closure{T,U} end
struct Pick{T} end
function muladd end

const A = Tuple{Tuple{Pick{1}, Closure{typeof(muladd), Tuple{Pick{2}, Pick{3}, Closure{typeof(muladd), Tuple{Pick{4}, Pick{5}, Closure{typeof(muladd), Tuple{Pick{6}, Pick{7}, Closure{typeof(muladd), Tuple{Pick{8}, Pick{9}, Closure{typeof(muladd), Tuple{Pick{10}, Pick{11}, Closure{typeof(muladd), Tuple{Pick{12}, Pick{13}, Closure{typeof(muladd), Tuple{Pick{14}, Pick{15}, Closure{typeof(muladd), Tuple{Pick{16}, Pick{17}, Closure{typeof(muladd), Tuple{Pick{18}, Pick{19}, Closure{typeof(muladd), Tuple{Pick{20}, Pick{21}, Closure{typeof(muladd), Tuple{Pick{22}, Pick{23}, Closure{typeof(muladd), Tuple{Pick{24}, Pick{25}, Closure{typeof(muladd), Tuple{Pick{26}, Pick{27}, Closure{typeof(muladd), Tuple{Pick{28}, Pick{29}, Closure{typeof(muladd), Tuple{Pick{30}, Pick{31}, Closure{typeof(muladd), Tuple{Pick{32}, Pick{33}, Closure{typeof(muladd), Tuple{Pick{34}, Pick{35}, Closure{typeof(muladd), Tuple{Pick{36}, Pick{37}, Closure{typeof(Base.:(*)), Tuple{Pick{38}, Pick{39}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}, Pick{40}}, Int64}
const B = Tuple{Tuple{Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(muladd), <:Tuple{Pick, Pick, Closure{typeof(Base.:(*)), <:Tuple{Pick, Pick}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}, Pick}, Int64}

println(A <: B) # takes a very, very long time... (>10 minutes)
```

This subtyping query is the root cause of the compilation blow-up in https://github.com/SciML/OrdinaryDiffEq.jl/issues/2535

### PR DESCRIPTION
**Pull request by N5N3**
close #56606

### CODE REVIEW

File: src/subtype.c [hunk 1]
```diff
@@ -1690,6 +1724,12 @@ static int forall_exists_equal(jl_value_t *x, jl_value_t *y, jl_stenv_t *e)
         }
     }
 
+    if (e->Loffset == 0 && jl_is_typevar(y) && jl_is_type(x) && (!e->intersection || !jl_is_uniontype(x))) {
+        // Fastpath for Type == TypeVar.
+        // Avoid duplicated `<:` check between adjacent `var_gt` and `var_lt`
```
Comment (N5N3) [inline src/subtype.c:1729]:
```suggestion
        // Avoid duplicated `<:` check between adjacent `var_gt` and `var_lt`
```

**Review by vtjnash — COMMENTED** [review]
LGTM

**Comment by N5N3** [comment]
@nanosoldier `runtests()`

**Comment by nanosoldier** [comment]
The package evaluation job [you requested](https://github.com/JuliaLang/julia/pull/56640#issuecomment-2493925993) has completed - possible new issues were detected.
The [**full report**](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/9087b94_vs_522f496/report.html) is available.

**Comment by N5N3** [comment]
@nanosoldier `runtests(["MethodInspector", "Algoim", "TuePlots", "AxisKeysExtra", "GLPK", "LazySets", "Optim", "SliceSampling", "LSODA", "OrdinaryDiffEqAdamsBashforthMoulton", "OrdinaryDiffEqNordsieck", "OrdinaryDiffEqLowOrderRK", "OrdinaryDiffEqStabilizedRK", "OrdinaryDiffEqQPRK", "OrdinaryDiffEqSSPRK", "OrdinaryDiffEqFeagin", "EconomicScenarioGenerators", "ParticleInCell", "GeometryOptimization", "FinanceModels", "FSimZoo", "BoundaryValueDiffEqMIRKN", "ODEProblemLibrary", "Yunir", "SymbolicInference", "FrequencySweep", "BatchReactor", "PlugFlowReactor", "SymbolicNumericIntegration"])`

**Comment by nsajko** [comment]
Backporting this to v1.11 seems to cause precompilation to stall for a specific package: #57152

**Pull request merged**